src/openforms/validations/registry.py

```diff
@@ -1,50 +1,32 @@
 import dataclasses
 import logging
-from typing import Callable, Iterable, List, Type, Union
+from typing import Iterable
 
 from django.core.exceptions import ValidationError as DJ_ValidationError
 from django.utils.translation import gettext_lazy as _
 
 import elasticapm
 from rest_framework import serializers
 from rest_framework.exceptions import ValidationError as DRF_ValidationError
+from typing_extensions import TypeVar
```

Comment:
can this not be imported from `typing`?

Reply:
Was experimenting with type var defaults. The base plugin now looks like:

```python
T = TypeVar("T", bound=JSONValue, default=str)

class BasePlugin(ABC, AbstractBasePlugin, Generic[T]):
    value_serializer: ClassVar[type[serializers.BaseSerializer]] = StringValueSerializer
```

As I'm giving a default serializer that works with strings, it makes sense to use `str` as a default to the type variable.

Let me know if you want plugins to be explicit with respect to their `value_serializer`

Reply:
The `default` argument was added, and is not available in `typing.TypeVar`

Reply:
I thought we'd only use features from `typing_extensions` that are available in newer versions of Python - this is exactly what I was worried about when I was pushing back against using `typing_extensions`.

The fact that it's being used by people is a non-argument - it's either putting the language itself in a hostage situation where they MUST adopt something because people started using it, or leads to non-standard behaviour, which in turn can interact negatively with other tooling (for example, the github code analysis recently had issues on some edge cases of pattern matching syntax, and that was on a core language feature, so you can expect non-standard things to lead to similar/worse issues).

I really like type annotations, but I'm also really concerned about the amount of effort and workarounds/coping strategies goes into applying them and making it work all in a complex project like Open Forms. Let's not lose focus of the actual features we're building.

Reply:
That's reasonable; however if we don't provide a default to the type var then we shouldn't provide a default value to the `value_serializer` imo, as it would lead to an inconsistent state between runtime and the type representation (even though nothing enforces a correct `value_serializer` to be set with respect to the solved type variable anyway)

Reply:
Please let's not get hung up on these typing things and focus on getting the code functional in the first place. I find it absurd that typing challenges (of which there are many because it's an evolving thing in the language) are limiting our DRY-ness of the actual runtime code.

By all means, apply them when it's low cost, but screw the typing information if it's not. There is too much energy going in these discussions that is better spent elsewhere.